### PR TITLE
Allow creation of ManagedMatrix without copy of vectors

### PR DESCRIPTION
**Pull request by blegat**
If we are able to directly generate these three vectors from the MOI wrapper and directly call the constructor, it would spare these copies.

### CODE REVIEW

File: src/types.jl [hunk 1]
```diff
@@ -60,6 +51,22 @@ struct ManagedSCSMatrix{T<:SCSInt}
...
+    values = convert(Vector{Cdouble}, copy(A.nzval))
+    rowval = convert(Vector{T}, A.rowval .- 1)
+    colptr = convert(Vector{T}, A.colptr .- 1)
+
+    return ManagedSCSMatrix(m, n, values, rowval, colptr)
+end
+
+function ManagedSCSMatrix{T}(m::Integer, n::Integer, A::Matrix) where T
+    return ManagedSCSMatrix(m, n, sparse(A))
+end
```
Comment (kalmarek) [inline src/types.jl:68]:
```suggestion
function ManagedSCSMatrix{T}(m::Integer, n::Integer, A::AbstractMatrix) where T
    return ManagedSCSMatrix{T}(m, n, sparse(A))
end
```
`Abstract` is not important, missing `T` is

File: src/types.jl [hunk 2]
```diff
@@ -60,6 +51,22 @@ struct ManagedSCSMatrix{T<:SCSInt}
...
+function ManagedSCSMatrix{T}(m::Integer, n::Integer, A::SparseMatrixCSC) where T
+    # we convert everything to make sure that no conversion will
+    # accidentally happen when ccalling `new`, so that `scsmatref`
+    # holds pointers to actual data stored in the fields.
+
+    values = convert(Vector{Cdouble}, copy(A.nzval))
+    rowval = convert(Vector{T}, A.rowval .- 1)
+    colptr = convert(Vector{T}, A.colptr .- 1)
+
+    return ManagedSCSMatrix(m, n, values, rowval, colptr)
```
Comment (kalmarek) [inline src/types.jl:63]:
```suggestion
    return ManagedSCSMatrix{T}(m, n, values, rowval, colptr)
```

**Comment by kalmarek** [comment]
btw. during the move of SCS to jump-dev I think I've lost access rights, i.e. I can't apply the suggestion

**Comment by mlubin** [comment]
@kalmarek Restored your access. It wasn't intentional.

**Comment by kalmarek** [comment]
@mlubin, no problem, thanks!

**Pull request merged**